source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ *  Copyright (C) 2011-2015  OpenDungeons Team
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <OgreVector3.h>
+#include <OgreCamera.h>
+
+#include "camera/CullingManager.h"
+#include "camera/CameraManager.h"
+#include "gamemap/GameMap.h"
+
+#include "entities/Creature.h"
+#include "utils/Vector3i.h"
+#include "utils/LogManager.h"
+#include <sstream>
+#include <algorithm>
+
+//! Values used to know whether to show and/or hide a mesh
+static const int HIDE =  1;
+static const int SHOW =  2;
+
+
+CullingManager::CullingManager(CameraManager* cameraManager):
+    mDebug(false),
+    mFirstIter(false),
+    mCm(cameraManager),
+    mCullCreaturesFlag(false),
+    mCullTilesFlag(false)
+{
+    mMyplanes =(Ogre::Plane(0, 0, 1, 0));
+    // init the Ogre vector
+    for (unsigned int i = 0; i < 8; ++i)
+    {
+        mOgreVectorsArray[i].x = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].y = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].z = static_cast<Ogre::Real>(0.0);
+    }
+}
+
+void CullingManager::cullTiles()
+{
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
```

Comment:
OD_LOG_INF is for LogMessageLevel::NORMAL
OD_LOG_DBG is for LogMessageLevel::TRIVIAL